### PR TITLE
Remove relay extension from AWS layer

### PR DESCRIPTION
**Pull request by sl0thentr0py**
we're reverting back to the older setup since the whole 'relay as AWS extension' experiment didn't really work out.

* remove gh action that bundles relay
* remove port override in DSN
* zip in place as part of `make build_aws_lambda_layer`

part of https://github.com/getsentry/team-webplatform-meta/issues/58

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,6 +20,7 @@ help:
 
 dist: .venv
 	rm -rf dist dist-serverless build
```
Comment (antonpirker) [inline Makefile:22]: It is just `/dist` now right?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -68,18 +68,6 @@ jobs:
           pip install virtualenv
           # This will also trigger "make dist" that creates the Python packages
           make aws-lambda-layer
-
-          echo "Saving SDK_VERSION for later"
-          export SDK_VERSION=$(grep "VERSION = " sentry_sdk/consts.py | cut -f3 -d' ' | tr -d '"')
-          echo "SDK_VERSION=$SDK_VERSION"
-          echo "SDK_VERSION=$SDK_VERSION" >> $GITHUB_ENV
-      - name: Upload Python AWS Lambda Layer
-        uses: getsentry/action-build-aws-lambda-extension@v1
```
Comment (antonpirker) [inline .github/workflows/ci.yml:77]: This gh action is also used in Javascript to create the excentions including relay: https://github.com/getsentry/sentry-javascript/blob/b9fb55f6c916ada89ec3c871078cc45661380f42/.github/workflows/build.yml#L285-L291

The idea behind this was to have one GH action to create the extension and include it in all the layers. 

So we can aim for two things now:
a) We remove this action in JS and remove also the actions repo as a whole. So going forward each language has to create the lambda layer on there own.
b) Keep the action as the single thing that takes a directory and makes a lambda layer out of it, so the different languages do not need to know how this is done. (having the layer zipping logic in one place only)

As without the lambda extension the action is only doing the zipping, i am fine with either option.

**Review by antonpirker — COMMENTED** [review]
See comments in "Conversation" tab.

File: Makefile (hunk 1 above)
Comment (sl0thentr0py) [reply Makefile:22]: yes

File: .github/workflows/ci.yml (hunk 2 above)
Comment (sl0thentr0py) [reply .github/workflows/ci.yml:77]: see TODO list in linked issue
https://github.com/getsentry/team-webplatform-meta/issues/58

File: scripts/aws-delete-lamba-layer-versions.sh [hunk 3]
```diff
@@ -8,7 +8,7 @@ set -euo pipefail
 # override default AWS region
 export AWS_REGION=eu-central-1
 
-LAYER_NAME=SentryPythonServerlessSDKLocalDev
+LAYER_NAME=SentryPythonServerlessSDK-local-dev
```
Comment (antonpirker) [inline scripts/aws-delete-lamba-layer-versions.sh:11]: Please check if the layer name is used also in the Sentry.io backend for setting up Lambda Functions integrations.

**Review by antonpirker — CHANGES_REQUESTED** [review]

File: scripts/aws-delete-lamba-layer-versions.sh (hunk 3 above)
Comment (sl0thentr0py) [reply scripts/aws-delete-lamba-layer-versions.sh:11]: the layer name is in craft
https://github.com/getsentry/sentry-python/blob/2bb95b3eaa9002f1367cd2600c33350b97aa8f8c/.craft.yml#L12

**Comment by bruceduhamel** [comment]
I came across this issue while investigating how to setup the aws lambda extension for some python lambdas I have integrated with sentry. 

My original goal was to improve lambda performance when events are delivered to sentry.io. 
I figured I would install the lambda extension via a lambda layer and configure my app to point to the relay.
I didn't find documentation for this approach, and this issue suggests that maybe isn't the way.

I found docs on how to setup a more point and click integration with lambda here: https://docs.sentry.io/product/integrations/cloud-monitoring/aws-lambda/#configure but these don't seem to have limited configuration support through a few environment variables, where I have specific config for the sentry python sdk and a few integrations that don't appear to be supported.

Is the lambda extension running relay no longer supported?

**Pull request merged**